ospfd/ospf_ri.c

```diff
@@ -2047,13 +2102,994 @@ DEFUN (no_pce_cap_flag,
 	return CMD_SUCCESS;
 }
 
-DEFUN (show_ip_ospf_router_info,
-       show_ip_ospf_router_info_cmd,
-       "show ip ospf router-info",
-       SHOW_STR
-       IP_STR
-       OSPF_STR
-       "Router Information\n")
+DEFPY(flex_algo_defn, flex_algo_defn_cmd, "flexible-algorithm (128-255)",
+      "Define a Flexible Algorithm Definition\n"
+      "Assign a unique number to the Flexible Algorithm\n")
+{
+	int id;
+	struct flex_algo_defn *fad;
+	bool new_fad = false;
+
+	if (!ospf_ri_enabled(vty))
+		return CMD_WARNING_CONFIG_FAILED;
+
+	id = atoi(argv[1]->arg);
+	if (id < 0 || id > FLEX_ALGO_ID_MAX)
```

Comment:
one suggestion : Here please make use of DEFPY usage by adding 'id' in command itself. ex :  flexible-algorithm (128-255)$id. So no need of atoi call here.  and also I believe DEFPY will take care of boundary validation, so no need of validation here.

Reply:
Thanks once again. Will change it.